jdaviz/app.py

```diff
@@ -887,6 +848,179 @@ def _get_all_subregions(mask, spec_axis_data):
 
         return regions
 
+    def get_subsets(self, subset_name=None, spectral_only=False,
+                    spatial_only=False, astropy_region=False):
+        """
+        Returns all branches of glue subset tree in the form that subset plugin can recognize.
+
+        Parameters
+        ----------
+        subset_name : str
+            The subset name.
+        spectral_only : bool
+            Return only spectral subsets.
+        spatial_only : bool
+            Return only spatial subsets.
+        astropy_region : bool
+            Return only astropy_region information and
+            leave out the region class name and glue_state.
+
+        Returns
+        -------
+        data : dict
+            A dict with keys representing the subset name and values as astropy regions
+            objects
+        """
+
+        dc = self.data_collection
+        subsets = dc.subset_groups
+        # TODO: Use global display units
+        # units = dc[0].data.coords.spectral_axis.unit
+        try:
+            viewer = self.get_viewer("spectrum-viewer")
+            units = u.Unit(viewer.state.x_display_unit) if viewer.state.x_display_unit else u.um
+        except Exception as e:
+            print(e)
```

Comment:
When I say "find correct units to use", this is what I mean. I know this is incorrect but a few alternatives I tried would not reliably get me the spectral axis unit for the data the subset applies to.